src/main/java/org/cirdles/topsoil/app/utils/SVGSaver.java

```diff
@@ -37,32 +43,74 @@
+                    writeSVGToOutputStream(svgDocument, stream);
+                });
+    }
+
+    /**
+     * Create a save dialog window and prompt user for filepath
+     *
+     * @return Optional<File>   the filepath
+     */
+    private Optional<OutputStream> generateSaveUI() {
         // get the path for the new SVG file
         FileChooser fileChooser = new FileChooser();
         fileChooser.setTitle("Export to SVG");
         fileChooser.setInitialDirectory(new File(System.getProperty("user.home")));
-        fileChooser.getExtensionFilters().add(new FileChooser.ExtensionFilter("SVG Image", "*.svg"));
-        File svgFile = fileChooser.showSaveDialog(null);
+        fileChooser.getExtensionFilters().setAll(
+                new FileChooser.ExtensionFilter("All Files", "*"),
+                new FileChooser.ExtensionFilter("SVG Image", "*.svg")
+        );
 
-        // can't convert if the file isn't specified
-        if (svgFile == null) {
-            return;
+        Optional<File> file = Optional.ofNullable(fileChooser.showSaveDialog(null));
+        if (file.isPresent()) {
+            try {
+                Optional<OutputStream> stream = Optional.ofNullable(new FileOutputStream(file.get()));
+                //Necessary for clean up
+                try {
+                    stream.get().close();
+                } catch (IOException ex) {
+                    Logger.getLogger(SVGSaver.class.getName()).log(Level.SEVERE, null, ex);
+                }
+                return stream;
+            } catch (FileNotFoundException ex) {
+                Logger.getLogger(SVGSaver.class.getName()).log(Level.SEVERE, null, ex);
+                return Optional.empty();
+            }
+        } else {
+            return Optional.empty();
         }
+    }
+
+    /**
+     * Writes the SVG file to the given path
+     *
+     * @param svgDocument   Vector image to be saved
+     * @param outStream     Given Path
+     */
+    void writeSVGToOutputStream(Document svgDocument, OutputStream outStream) {
 
         // write the SVG to the given path
         try {
-            Transformer transformer = TransformerFactory.newInstance()
-                    .newTransformer();
+            Transformer transformer = TransformerFactory.newInstance().newTransformer();
 
             // configure the resulting SVG document
             transformer.setOutputProperty(OutputKeys.DOCTYPE_PUBLIC, SVG_DOCTYPE_PUBLIC);
             transformer.setOutputProperty(OutputKeys.DOCTYPE_SYSTEM, SVG_DOCTYPE_SYSTEM);
             transformer.setOutputProperty(javax.xml.transform.OutputKeys.INDENT, "yes");
 
-            transformer.transform(new DOMSource(svgDocument), new StreamResult(svgFile));
-        } catch (TransformerConfigurationException ex) {
-            Logger.getLogger(SVGSaver.class.getName()).log(Level.SEVERE, null, ex);
+            transformer.transform(new DOMSource(svgDocument), new StreamResult(outStream));
```

Comment:
I think you need to close the outputStream after this. Put it in a `finally` block.